soroban-env-host/src/host.rs

```diff
@@ -2087,11 +2108,37 @@ impl VmCallerEnv for Host {
         self.extend_contract_instance_and_code_ttl_from_contract_id(
             &contract_id,
             threshold.into(),
-            extend_to.into(),
+            extend_to.into()
         )?;
         Ok(Val::VOID)
     }
 
+    fn extend_contract_code_ttl(
+        &self,
+        _vmcaller: &mut VmCaller<Self::VmUserState>,
+        contract: AddressObject,
+        threshold: U32Val,
+        extend_to: U32Val,
+    ) -> Result<Void, Self::Error> {
+        let contract_id = self.contract_id_from_address(contract)?;
+        let key = self.contract_instance_ledger_key(&contract_id)?;
+
+        match self
+            .retrieve_contract_instance_from_storage(&key)?
+            .executable
+        {
+            ContractExecutable::Wasm(wasm_hash) => {
+                let key = self.contract_code_ledger_key(&wasm_hash)?;
+                self.try_borrow_storage_mut()?
+                    .extend_ttl(self, key, threshold.into(), extend_to.into())
+                    .map_err(|e| self.decorate_contract_code_storage_error(e, &wasm_hash))?;
+            }
+            ContractExecutable::StellarAsset => {}
+        }
+
+        Ok(Val::VOID)
+    }
+
```

Comment:
Nit: I notice this code (and one above) is duplicate of `extend_contract_instance_and_code_ttl_from_contract_id`. Would you mind a minor refactoring to avoid duplication? The logic here isn't too complex so it isn't a huge deal, it's just for better maintainability. 
You can just add a new helper in `data_helper` to extend the code, and change `extend_contract_instance_and_code_ttl_from_contract_id` to only extend contract instance (or vice versa).